plugins/container_logs.yaml

```diff
@@ -105,6 +109,15 @@ template: |
             id: log-to-body
             from: attributes.log
             to: body
+
+          # Parse body as json if the body appears to be json
+          {{ if .body_json_parsing }}
```

Comment:
I've been struggling with the idea of boolean flags for JSON vs Regex parsing for a while now. On one hand it probably is more performant than checking format of the log entry to see if it is JSON vs a regex-able log entry, however I think from a usability perspective it suffers as its another barrier for users to start using the plugin. Also if you have some pods use JSON logging and some use regex then the container logs are already incompatible and would need to be another plugin configuration etc. 

I'd like other perspectives but I'm of the camp that we emulate something like the mongodb plugin or we don't have the boolean parameter. 
https://github.com/observIQ/observiq-otel-collector/blob/fb99ec8ecf684a45a8898319c811e491bbcba21c/plugins/mongodb_logs.yaml#L41-L45

I don't want to be the only person to dictate that as a choice though, so I'm open to having a larger discussion.

Reply:
I don't want to diminish the performance impact but it is something that I've been thinking about quite a bit as far as headaches to start using plugins. For this case in a containerized environment it may be more complex to manually modify the plugins to optimize performance, so I can see that as an argument to keep the boolean flag. 

TLDR; think this is might be fine but just thought I'd share some of my internal conflictions/consistency with some of the boolean flag design 🙃

Reply:
I think this case is a little odd/different.

This isn't really a regex parsing vs JSON parsing thing; There doesn't seem to be any further regex parsing, this is more just a toggle for optional extra parsing of the log message, if the log message happens to be JSON (IDK if this is often the case or not). It is also conditional similar to your example, when enabled (`if: body matches "^{.*}\\s*$"`).

I like the idea of being able to disable this if you want to squeak out extra performance and you know that your logs aren't JSON.

Reply:
The json_parser's `if` condition will allow the operator to be skipped in the event that a log is not json. The only downside here is the clear performance penalty of checking every line before parsing or skipping.

From a usability perspective, there is no prerequisite for the user. They can have a mixed environment without running into issues.

Users that do not have json logs (or do not wish to parse the json logs) can simply disable this option. 

While testing this change, I used Google's microservice demo, which has a handful of services with differing log formats. I did not observe parsing errors, but I did observe all json logs being parsed to body.

Reply:
> I like the idea of being able to disable this if you want to squeak out extra performance and you know that your logs aren't JSON.

I get that so I'll resolve this. Just in general I was thinking of the concept of whether nested JSON or in general how we should approach it. I see now its a slippery spiral and this is a tad niche of a use case so I'll go ahead and resolve this though